### ISSUE TITLE
USWDS - Bug: Fix invalid aria role in validator list items

### ISSUE THREAD

**Issue by mejiaj** [uswds/uswds#4905]
### Describe the bug

Testing validator updates in uswds/uswds#4719 results in the following a11y errors:

```bash
Errors in http://localhost:4000/components/validation/:

 • Elements must only use allowed ARIA attributes
   (https://dequeuniversity.com/rules/axe/3.5/aria-allowed-attr?application=axeAPI)

   (#validate-code > li:nth-child(1))

   <li class="usa-checklist__item" data-validator="uppercase"
   aria-checked="false"> Use at least one u...</li>

 • Elements must only use allowed ARIA attributes
   (https://dequeuniversity.com/rules/axe/3.5/aria-allowed-attr?application=axeAPI)

   (#validate-code > li:nth-child(2))

   <li class="usa-checklist__item" data-validator="numerical"
   aria-checked="false"> Use at least one n...</li>
```

### Steps to reproduce the bug

1. Use branch for uswds/uswds#4719
2. Build site
3. Create federalist preview with draft PR
5. View failed CircleCI tests
6. Click `details` link
7. See errors 
![image](https://user-images.githubusercontent.com/3385219/183151976-89ad103d-28c7-4384-acc2-77ac94573053.png)

### Expected Behavior

A11Y tests should pass

### Related code

Work was done in USWDS PR uswds/uswds#4719.

### System setup

- USWDS v3.1.0

### Additional context

We should look into adding the necessary role 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked#associated_roles

Or change the markup so it the validation state is accurate without `aria-checked`

**After adding `aria-role="checkbox"`**

![image](https://user-images.githubusercontent.com/3385219/183156765-23fbb402-d359-4027-81b8-00c60357adcf.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.
